### ISSUE TITLE
WakeOnLAN wakes Laptop, but it keeps waking every few minutes despite checking "Only allow magic packet to wake.."

### ISSUE THREAD

**Issue by drb01** [basildane/WakeOnLAN#194]
WakeOnLAN works well for remotely waking my Win 10 64bit laptop.  However for it to work I have to check "Allow this device to wake the computer" on the WiFi adapter. I also check "Only allow magic packets...". The problem is that the laptop just wakes up every few minutes anyway (which it doesn't do if these are unchecked). 

I have tried Tools->Listener which shows the packets e.g. if I send a WOL from my phone. But it shows nothing when the laptop is woken, or if I just leave it on monitoring the network. 

Another ethernet wired PC on the same network works perfectly. WOL wakes it, and it stays asleep in the meantime.

powercfg -lastwake shows that it is the laptop's WiFi card waking it.

Any idea what is waking the laptop every few minutes?

**Comment by basildane** [comment]
I have never encountered this.  I will do some research, but I don't have any quick answer.

**Comment by drb01** [comment]
I've tried a few things to no avail:
- reinstall the driver for the WiFi card
- try old Win 7 driver
- uninstall VPN
- Safe Mode (doesn't have Sleep mode)
- Hibernate - still wakes up!
- disable Fast Boot in BIOS

The only thing that worked, unsurprisingly, is disabling the WiFi card in Device Manger.  If I use a USB Ethernet  connection then it sleeps & WOL as expected. 

Any help very much appreciated!

**Comment by basildane** [comment]
What is the model of the wifi adapter?  Is it made by Intel?

**Comment by drb01** [comment]
Sure is - "Intel(R) Dual Band Wireless-AC 8260"

It's in an Asus Zenpad UX530 which is about 4.5 years old and works very well.

**Comment by basildane** [comment]
Good choice.  Have you checked Advanced properties, specifically make sure that Wake on Pattern Match and other things like that are disabled.

**Comment by basildane** [comment]
Also, do you have the Proset drivers installed?

**Comment by drb01** [comment]
I have looked in ADvanced Properties but not much of it makes sense. I have both Wake on Pattern Match and Wake on Magic Packet enabled, otherwise WOL doesn't work.

I have tried Intel's driver (Currently 20.70.13.2) and I tried a Microsoft driver too. And an old Win 7 driver from 2017 - they were all the same. 

I have not tried a Proset driver (never heard of it before). Just found it online so I'll give it a go.

**Comment by basildane** [comment]
I would disable pattern match

**Comment by drb01** [comment]
I installed the v22.120.1 Proset drivers although in Device Manager it says v20.70.30.1 is installed, dated 13 Jan 2022. I tried installing several times. The previous driver was dated 2019 so it's a step in the right direction.

In Advanced, Wake on Pattern is disabled, Wake on Magic is enabled. In the Power Management Tab, Allow this device to wake the computer is enabled, and Only allow a Magic Packet is also enabled.

Still getting the same problem :(

**Comment by drb01** [comment]
Does anyone have any other ideas on this?

**Comment by basildane** [comment]
Does the laptop wake up, even if the WOL application is not running at all?

**Comment by BWolfer54** [comment]
If there's a property 'wake on link', please try disabling it, too.

**Comment by basildane** [comment]
> If there's a property 'wake on link', please try disabling it, too.

Yes, Wake on Magic Packet should be the only option enabled.  Link change, pattern match, and others could cause this.
Do you have those other options?  Are they disabled?

**Comment by drb01** [comment]
Unfortunately I can't find a Wake on Link option. I have always been sure to keep Wake on Magic Packet enabled...

![WOL](https://user-images.githubusercontent.com/25118781/167502793-587942e5-186d-45bb-ab21-846259a8eb99.jpg)
![WOL1](https://user-images.githubusercontent.com/25118781/167502966-dc73ff20-aeb1-4e26-bf7d-4126b2e15b05.jpg)

**Comment by Moopere** [comment]
I've never seen wifi based WOL work reliably and mostly it doesn't seem to work at all ime.  I think you've done really well to even get this far.

I suspect the problem you are seeing is the wifi card triggering the laptop to wake up to check if it should be getting woken up :)   Most likely the WOL feature for this laptop, such as it is, was only designed to work from standby rather than G3 shutdown.

What I generally do for laptops is use a docking station with wired ethernet.  Still plenty of compatibility problems with USB style docks - many laptops just don't have the firmware to properly respond to USB events when the laptop is G3/Turned off.
